contracts/smart-account/utils/AddressResolver.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {IEcdsaOwnershipRegistryModule} from "../interfaces/modules/IEcdsaOwnershipRegistryModule.sol";
+import {IAddressResolver} from "../interfaces/IAddressResolver.sol";
+import {ISmartAccount} from "../interfaces/ISmartAccount.sol";
+
+contract AddressResolver {
+    struct SmartAccountResult {
```

Comment:
should be fine imo.
the interfaces I am using are not being implemented as they have other methods also.

Reply:
if you notice AddressResolver is ! IAddressResolver here. I could write two different interfaces. one being this including struct and other one renamed to ISmartAccountFactory interfaces (cant reuse them as for factory we did not create interfaces)

Reply:
> if you notice AddressResolver is ! IAddressResolver here.

Didn't get this

Reply:
> I could write two different interfaces. one being this including struct and other one renamed to ISmartAccountFactory interfaces (cant reuse them as for factory we did not create interfaces)

Why do we need two different interfaces?

Reply:
AddressResolver does not implement IAddressResolver. which has factory deployment method interface
 
 and factory interface I just figured we have for V2 factory
 before I combined V1 and V2 factory interfaces and named it IAddressResolver which is wrong. 
 
 second interface I mean actual interface where this struct would be moved.
 
 I am pushed updated changes, check them it would make more sense.

Reply:
depending on design decisions in the sdk code may change for this contract so my priority is to get mvp out and get it tested. so I would appreciate if rules are not strict - at this stage for such utility contract

Reply:
> AddressResolver does not implement IAddressResolver. which has factory deployment method interface
> 
> and factory interface I just figured we have for V2 factory before I combined V1 and V2 factory interfaces and named it IAddressResolver which is wrong.
> 
> second interface I mean actual interface where this struct would be moved.
> 
> I am pushing updated changes, check them it would make more sense.

https://github.com/bcnmy/scw-contracts/commit/da16499ce8103f43d2a19b759c3095ac4e0d9d22

Reply:
Agreed with keeping things simple, but I think if a change is fairly trivial and helps with consistency and technical debt, we should do the changes.

Reply:
conflict in AddressResolver not implementing interface with same name seemed trivial to me. so I realised and fixed it. In that name some other interface was being used so it makes sense.
moving struct is unnecessary ask imo. anyway I have pushed